posthog/api/insight.py

```diff
@@ -224,7 +241,29 @@ def get_effective_privilege_level(self, insight: Insight) -> Dashboard.Privilege
 
     def to_representation(self, instance: Insight):
         representation = super().to_representation(instance)
-        representation["filters"] = instance.dashboard_filters(dashboard=self.context.get("dashboard"))
+
+        dashboard: Optional[Dashboard] = self.context.get("dashboard")
+        representation["filters"] = instance.dashboard_filters(dashboard=dashboard)
+        representation = self._choose_filters_hash(dashboard, instance, representation)
+
+        return representation
+
+    def _choose_filters_hash(self, dashboard, instance, representation):
+        if dashboard is not None:
+            # noinspection PyBroadException
+            try:
+                dashboard_context_filters_hash = DashboardTile.objects.get(
+                    dashboard__id=dashboard.id, insight=instance
+                ).filters_hash
+                representation["filters_hash"] = dashboard_context_filters_hash
+            except Exception as e:
+                logger.error(
+                    "insight_on_dashboard.could_not_load_filters_hash",
+                    insight=instance.id,
+                    dashboard=dashboard.id,
+                    exception=e,
+                )
+
         return representation
```

Comment:
hmmm, I was giving that method a surface that hid that it was coupled to a specific instance cos it is yucky to act at a distance.

I'll copy the structure so that the contract is more strict

Reply:
actually, maybe we don't need the method at all 🤔